packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -3907,14 +3951,17 @@ export function hoistPreambleState(
   preambleState: PreambleState,
 ) {
   const rootPreamble = renderState.preamble;
-  if (rootPreamble.htmlChunks === null) {
+  if (rootPreamble.htmlChunks === null && preambleState.htmlChunks) {
     rootPreamble.htmlChunks = preambleState.htmlChunks;
+    preambleState.contribution |= HTMLContribution;
```

Comment:
I guess the `contribution` flag means that the chunks of this preamble were handled and passed along? I was expecting this to set the `contribution` flag on the `rootPreamble` instead.

Reply:
Yeah the reason we even need to track which boundary contributed to which part of the preamble is because if that boundary ever needs to be retried on the client we need to clean up the disconnected parts since they won't be 'inside' that boundary like normal tags would be. The root never gets cleaned up (except in that the entire container might be cleared but that will already be the entire Document if you are rendering these singletons anyway) so when a Suspense boundary is contributing a part of the preamble it tracks in it's own state that it's preamble parts were actually used so we can associate the necessary info with that boundary on the client to clean it up.